core/operation/src/main/java/uk/gov/gchq/gaffer/operation/Operation.java

```diff
@@ -28,7 +30,12 @@
  * Operations must be JSON serialisable in order to make REST API calls.
  */
 @JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.PROPERTY, property = "class")
-public interface Operation {
+public interface Operation extends Closeable {
+
+    default void close() throws IOException {
```

Comment:
I'm not sure what the best solution is. As it is all operations are closeable and users can chose to override the close method.

It also does makes sense for an operation implementation to chose whether to implement closable or not and add "implements Closeable". 

I think making all operations implement closeable will encourage more users to think about adding a custom implementation of close.

It also means in OperationChain.close I can just iterate round all operations and call close. Alternatively I would need to do instanceof and then cast to Closeable for each operation.